src/Tribe/Main.php

```diff
@@ -566,40 +566,50 @@ public function should_load_common_admin_css() {
 	}
 
 	/**
-	 * A Helper method to load text domain
-	 * First it tries to load the wp-content/languages translation then if falls to the try to load $dir language files.
+	 * A helper method to load text domain.
+	 * First, it tries to load language files from a custom folder when set. If it fails it falls back to load from $dir.
 	 *
 	 * @since  4.0.1 Introduced.
 	 * @since  4.2   Included $domain and $dir params.
+	 * @since  TBD   Fix text domain loading logic after changes in WordPress 6.7.
 	 *
 	 * @param string       $domain The text domain that will be loaded.
 	 * @param string|false $dir    What directory should be used to try to load if the default doesn't work.
 	 *
 	 * @return bool  If it was able to load the text domain.
 	 */
 	public function load_text_domain( $domain, $dir = false ) {
-		// Added safety just in case this runs twice...
+		// Added safety just in case this runs twice.
 		if ( is_textdomain_loaded( $domain ) && ! $GLOBALS['l10n'][ $domain ] instanceof NOOP_Translations ) {
 			return true;
 		}
 
 		$locale = get_locale();
-		$plugin_rel_path = WP_LANG_DIR . '/plugins/';
+		$plugin_rel_path = $dir;
 
 		/**
-		 * Allows users to filter the file location for a given text domain..
-		 * Be careful when using this filter, it will apply across the whole plugin suite.
+		 * Allows users to filter the file location for a given text domain.
+		 * Be careful when using this filter, it will apply across the whole plugin suite when not checking $domain.
 		 *
 		 * @param string      $plugin_rel_path The relative path for the language files.
-		 * @param string      $domain Which plugin domain we are trying to load.
-		 * @param string      $locale Which Language we will load.
-		 * @param string|bool $dir    If there was a custom directory passed on the method call.
+		 * @param string      $domain          The plugin domain we are trying to load.
+		 * @param string      $locale          The language we will load.
+		 * @param string|bool $dir             If there was a custom directory passed on the method call.
 		 */
 		$plugin_rel_path = apply_filters( 'tribe_load_text_domain', $plugin_rel_path, $domain, $locale, $dir );
 
+		// Try to load translations from the plugin's language folder or a custom folder.
 		$loaded = load_plugin_textdomain( $domain, false, $plugin_rel_path );
```

Comment:
I would remove this and at the end of the method just return `load_plugin_textdomain( $domain, false, $plugin_rel_path )`

Reply:
@codingmusician @dpanta94 (your comment about moving above line 602).
We offer a filter on line 599 for the user to define a custom folder, so we need to try loading that textdomain first.
If the user's custom folder fails, then we still try the plugin's lang folder.
If it's not filtered, then `$dir` - which is the plugin's lang folder usually (coming from us) - goes into line 602. Then the check on line 608 fails due to the last condition and is not executed anymore.
Also, the minimal changes offers easy backwards-compatibility.

And I agree, the `$loaded =` can be removed.